Add documentation on common exit statuses

roclark opened the pull request:
Potential non-zero exit status need to be documented properly to give users an idea of what went wrong and how to resolve any issues. A dedicated troubleshooting guide is added to compliment this work.

Fixes #9 

Signed-Off-By: Robert Clark <roclark@nvidia.com>

roclark committed:
Add specificity to exit codes

The application used generic codes without any specificity, making it
hard to discerne what went wrong based on the exit code. Adding
specificity makes it easier to users to troubleshoot an issue when
things don't go as expected.

Signed-Off-By: Robert Clark <roclark@nvidia.com>

roclark committed:
Add a troubleshooting guide

The troubleshooting guide helps users identify common issues and how to
resolve them.

Signed-Off-By: Robert Clark <roclark@nvidia.com>

joehandzik commented on bobber/lib/exit_codes.py:
```diff
@@ -0,0 +1,6 @@
+# SPDX-License-Identifier: MIT
+# This file contains a list of exit codes for debugging
+SUCCESS = 0  # Successful termination
+BASELINE_FAILURE = 10  # Performance did not meet criteria
```
Should failure conditions be negative, rather than positive?

roclark replied:
Good idea, I can update that.

roclark commented:
Updated the exit codes and changed the doc a bit to reflect the latest updates to the repo. I included a bit of space in the exit codes to group together similar items. For example, future updates will include more error handling for Docker and I plan on having those codes in the -30s while future parser errors will land in the -20s.
